webui/react/src/components/ResourcePoolDetails.tsx

```diff
@@ -16,15 +16,14 @@ interface Props {
 }
 
 const ResourcePoolDetails: React.FC<Props> = ({ resourcePool: pool, ...props }: Props) => {
-  const details = clone(pool.details);
+  const details = structuredClone(pool.details);
   for (const key in details) {
-    if (details[key] === null) {
-      delete details[key];
+    if (details[key as keyof V1ResourcePoolDetail] === null) {
+      delete details[key as keyof V1ResourcePoolDetail];
     }
   }
 
-  const mainSection = clone(pool);
-  delete mainSection.details;
+  const mainSection = structuredClone(pool);
```

Comment:
given that `structuredClone` works as a deep clone, couldn't you
1) clone pool
2) save a reference to the cloned pool's details for details
3) then delete `details` from the cloned pool
?

Reply:
or consider:
```tsx
const { details, ...mainSection } = pool;
```